pkg/executor/importer/import.go

```diff
@@ -1181,12 +1204,43 @@ func (e *LoadDataController) InitDataFiles(ctx context.Context) error {
 			return err
 		}
 	}
+	if err2 = e.checkNonCSVFormatOptions(isDataFormatAuto); err2 != nil {
+		return err2
+	}
 
 	e.dataFiles = dataFiles
 	e.TotalFileSize = totalSize
 	return nil
 }
 
+// update format of the validated file by its extension.
+func (e *LoadDataController) detectAndUpdateFormat(path string) {
+	if e.Format == DataFormatAuto {
+		e.Format = parseFileType(path)
+		e.logger.Info("detect and update import plan format based on file extension", zap.String("file", path), zap.String("detected format", e.Format))
+		e.Parameters.Format = e.Format
+	}
+}
+
+func parseFileType(path string) string {
+	path = strings.ToLower(path)
+	ext := filepath.Ext(path)
+	// avoid duplicate compress extension
+	for ext == ".gz" || ext == ".gzip" || ext == ".zstd" || ext == ".zst" || ext == ".snappy" {
```

Comment:
```suggestion
	if ext == ".gz" || ext == ".gzip" || ext == ".zstd" || ext == ".zst" || ext == ".snappy" {
```
we don't support multiple compression types

Reply:
fixed and remove multiple compression types relate tests